process_report/tests/e2e/README.md

```diff
@@ -0,0 +1,121 @@
+# End-to-End (E2E) Tests
+
+This directory contains end-to-end tests for the NERC invoicing pipeline. These tests are designed to validate that the entire pipeline runs successfully from start to finish with realistic data.
+
+## Purpose
+
+The E2E tests are **not** intended to cover all edge cases or detailed validation - that's what unit tests are for. Instead, they:
+
+1. **Validate Pipeline Execution**: Ensure the full pipeline runs without errors
+2. **Check Basic Outputs**: Verify that all expected output files are generated
+3. **Structural Validation**: Perform basic sanity checks on output file structures
+4. **Integration Testing**: Test that all pipeline components work together correctly
+
+## What the E2E Test Covers
+
+The main E2E test (`test_e2e_pipeline.py`) tests the following pipeline stages:
+
+### Input Processing
+- Combines multiple CSV invoice files (OpenShift, OpenStack, Storage)
+- Passing all inputs to `Processor` and `Invoice` subclasses
+
+## Test Data
+
+The `test_data/` directory contains the minimal dataset required by the pipeline.
+
+- **CSV Invoice Files**: `test_invoice_openshift.csv`, `test_invoice_openstack.csv`, `test_invoice_storage.csv`
+- **Configuration Files**: Non-billable PIs/projects, timed projects, prepay configurations
+- **Historical Data**: Old PI file, alias mappings, prepay debits
+
+## Running the Tests
+
+### Prerequisites
+
+1. Install dependencies:
+   ```bash
+   pip install -r requirements.txt
+   ```
+
+2. Install Chromium (for PDF generation, though it's mocked in tests):
+   ```bash
+   # Ubuntu/Debian
+   sudo apt-get install chromium-browser
+
+   # macOS
+   brew install chromium
+   ```
+
+### Local Execution
+
+Run the E2E tests locally:
+
+```bash
+# From the project root directory
+python -m unittest discover -s process_report/tests/e2e
+
+# Or run the specific test file
+python -m unittest process_report.tests.e2e.test_e2e_pipeline
```

Comment:
These 2 command didn't run the tests. `test_e2e_pipeline.py` does not use the unittest module. The tests need to be in a class that inherit from the base unittest.TestCase. 

You should just document that you expect this to run with pytest (as you are doing it in the github workflows) and don't worry about unittest at all.